torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -0,0 +1,417 @@
+        "Invert": lambda input, magnitude, interpolation, fill: F.invert(input),
+    }
+
+    def _is_supported(self, obj: Any) -> bool:
+        return type(obj) in {features.Image, torch.Tensor} or isinstance(obj, PIL.Image.Image)
+
+    def _get_params(self, sample: Any) -> Dict[str, Any]:
+        image: Any = next(
+            query_recursively(
+                lambda input: input if self._is_supported(input) else None, sample  # type: ignore[no-any-return]
+            )
+        )
+        num_channels = F.get_image_num_channels(image)
+
+        fill = self.fill
+        if isinstance(fill, (int, float)):
+            fill = [float(fill)] * num_channels
+        elif fill is not None:
+            fill = [float(f) for f in fill]
+
+        return dict(interpolation=self.interpolation, fill=fill)
+
+
+@dataclasses.dataclass
+class AugmentationMeta:
+    dispatcher_id: str
+    magnitudes_fn: Callable[[int, Tuple[int, int]], Optional[torch.Tensor]]
+    signed: bool
+
+
+class AutoAugment(_AutoAugmentBase):
+    _AUGMENTATION_SPACE = (
+        AugmentationMeta("ShearX", lambda num_bins, image_size: torch.linspace(0.0, 0.3, num_bins), True),
+        AugmentationMeta("ShearY", lambda num_bins, image_size: torch.linspace(0.0, 0.3, num_bins), True),
+        AugmentationMeta(
+            "TranslateX",
+            lambda num_bins, image_size: torch.linspace(0.0, 150.0 / 331.0 * image_size[1], num_bins),
+            True,
+        ),
+        AugmentationMeta(
+            "TranslateY",
+            lambda num_bins, image_size: torch.linspace(0.0, 150.0 / 331.0 * image_size[0], num_bins),
+            True,
+        ),
+        AugmentationMeta("Rotate", lambda num_bins, image_size: torch.linspace(0.0, 30.0, num_bins), True),
+        AugmentationMeta("Brightness", lambda num_bins, image_size: torch.linspace(0.0, 0.9, num_bins), True),
+        AugmentationMeta("Color", lambda num_bins, image_size: torch.linspace(0.0, 0.9, num_bins), True),
+        AugmentationMeta("Contrast", lambda num_bins, image_size: torch.linspace(0.0, 0.9, num_bins), True),
+        AugmentationMeta("Sharpness", lambda num_bins, image_size: torch.linspace(0.0, 0.9, num_bins), True),
+        AugmentationMeta(
+            "Posterize",
+            lambda num_bins, image_size: cast(torch.Tensor, 8 - (torch.arange(num_bins) / ((num_bins - 1) / 4)))
+            .round()
+            .int(),
+            False,
+        ),
+        AugmentationMeta("Solarize", lambda num_bins, image_size: torch.linspace(255.0, 0.0, num_bins), False),
+        AugmentationMeta("AutoContrast", lambda num_bins, image_size: None, False),
+        AugmentationMeta("Equalize", lambda num_bins, image_size: None, False),
+        AugmentationMeta("Invert", lambda num_bins, image_size: None, False),
+    )
+    _AUGMENTATION_SPACE = {
+        augmentation_meta.dispatcher_id: augmentation_meta for augmentation_meta in _AUGMENTATION_SPACE
```

Comment:
Nit: Why don't we explicitly define the map similar to what we do at `_DISPATCHER_MAP`?

Note that the 2 share the exact same keys. In this case, you could remove the `id` from `AugmentationMeta` and rename it Magnitudes, though TBH I'm not sure why a simple tuple son't do in this case aka:
str: (lambda, bool)